src/Schema.js

```diff
@@ -409,6 +409,88 @@ Schema.prototype.validateField = function(className, key, type, freeze) {
   });
 };
 
+// Delete a field, and remove that data from all objects. This is intended
+// to remove unused fields, if other writers are writing objects that include
+// this field, the field may reappear. Returns a Promise that resolves with
+// no object on success, or rejects with { code, error } on failure.
+
+// Passing the database and prefix is necessary in order to drop relation collections
+// and remove fields from objects. Ideally the database would belong to
+// a database adapter and this fuction would close over it or access it via member.
+Schema.prototype.deleteField = function(fieldName, className, database, prefix) {
+  if (!classNameIsValid(className)) {
+    return Promise.reject({
+      code: Parse.Error.INVALID_CLASS_NAME,
+      error: invalidClassNameMessage(className),
+    });
+  }
+
+  if (!fieldNameIsValid(fieldName)) {
+    return Promise.reject({
+      code: Parse.Error.INVALID_KEY_NAME,
+      error: 'invalid field name: ' + fieldName,
+    });
+  }
+
+  //Don't allow deleting the default fields.
+  if (!fieldNameIsValidForClass(fieldName, className)) {
```

Comment:
That seems plausible. Next time I'm writing validation code I'll try that style and see if I like it better.